modules/dns_dnscmd/dns_dnscmd_main.rb

```diff
@@ -87,15 +97,24 @@ def report msg, response, error_only
       raise Proxy::Dns::Error.new("Unknown error while processing '#{msg}'")
     end
 
-    def match_zone(fqdn)
+    def match_zone(fqdn_or_ip)
```

Comment:
Sorry, not getting that. How would I pass the zone name if it is the method's job to determine exactly that?
I rather like the idea to be able to pass an IP or FQDN.